ignite/src/main/java/org/hibernate/ogm/datastore/ignite/IgniteDialect.java

```diff
@@ -0,0 +1,780 @@
+					result = (long) request.getInitialValue();
+					if ( !cache.putIfAbsent( idSourceKey, result ) ) {
+						previousValue = (long) request.getInitialValue();
+					}
+				}
+				if ( previousValue != null ) {
+					while ( true ) {
+						result = previousValue + request.getIncrement();
+						if ( cache.replace( idSourceKey, previousValue, result ) ) {
+							break;
+						}
+						else {
+							previousValue = cache.get( idSourceKey );
+						}
+					}
+				}
+				break;
+			case SEQUENCE:
+				IgniteAtomicSequence seq = provider.atomicSequence( request.getKey().getMetadata().getName(), request.getInitialValue(), true );
+				result = seq.getAndAdd( request.getIncrement() );
+				break;
+		}
+		return result;
+	}
+
+	@Override
+	public boolean supportsSequences() {
+		return true;
+	}
+
+	@Override
+	public void forEachTuple(ModelConsumer consumer, TupleTypeContext tupleTypeContext, EntityKeyMetadata entityKeyMetadata) {
+		throw new UnsupportedOperationException( "forEachTuple() is not implemented" );
+	}
+
+	@Override
+	public int executeBackendUpdateQuery(BackendQuery<IgniteQueryDescriptor> query, QueryParameters queryParameters, TupleContext tupleContext) {
+		throw new UnsupportedOperationException( "executeBackendUpdateQuery() is not implemented" );
+	}
+
+	@Override
+	public ClosableIterator<Tuple> executeBackendQuery(BackendQuery<IgniteQueryDescriptor> backendQuery, QueryParameters queryParameters,
+			TupleContext tupleContext) {
+		IgniteCache<Object, BinaryObject> cache;
+		if ( backendQuery.getSingleEntityMetadataInformationOrNull() != null ) {
+			cache = provider.getEntityCache( backendQuery.getSingleEntityMetadataInformationOrNull().getEntityKeyMetadata() );
+		}
+		else if ( backendQuery.getQuery().getQuerySpaces().size() > 0 ) {
+			cache = provider.getEntityCache( backendQuery.getQuery().getQuerySpaces().iterator().next() );
+		}
+		else {
+			throw new IgniteHibernateException( "Can't find cache name" );
+		}
+		QueryHints hints = ( new QueryHints.Builder( queryParameters.getQueryHints() ) ).build();
+		SqlFieldsQuery sqlQuery = provider.createSqlFieldsQueryWithLog(
+				backendQuery.getQuery().getSql(),
+				hints,
+				IgniteHqlQueryParser.createParameterList( backendQuery.getQuery().getOriginalSql(), queryParameters.getNamedParameters() ).toArray() );
+		Iterable<List<?>> result = executeWithHints( cache, sqlQuery, hints );
+
+		if ( backendQuery.getQuery().isHasScalar() ) {
+			return new IgniteProjectionResultCursor( result, backendQuery.getQuery().getCustomQueryReturns(), queryParameters.getRowSelection() );
+		}
+		else {
+			if ( backendQuery.getSingleEntityMetadataInformationOrNull() != null ) {
+				return new IgnitePortableFromProjectionResultCursor(
+								result,
+								queryParameters.getRowSelection(),
+								backendQuery.getSingleEntityMetadataInformationOrNull().getEntityKeyMetadata()
+							);
+			}
+			else {
+				throw new UnsupportedOperationException( "Not implemented yet" );
```

Comment:
I think we might actually have a SQL parser already implemented, I need to check the project as I'm not sure and I cannot find it right now. I'll have a look tomorrow